File: frontend/src/types/appointment.ts
export type AppointmentStatus =
  | "scheduled"
  | "completed"
  | "cancelled";

export interface Appointment {
  id: number;
  title: string;
  description: string | null;
  appointment_date: string;
  start_time: string;
  end_time: string;
  status: AppointmentStatus;
  created_at: string;
  updated_at: string;
}

export interface AppointmentCreate {
  title: string;
  description?: string;
  appointment_date: string;
  start_time: string;
  end_time: string;
}

export interface AppointmentUpdate {
  title?: string;
  description?: string;
  appointment_date?: string;
  start_time?: string;
  end_time?: string;
}

File: backend/app/schemas/appointment.py
from datetime import date, time, datetime
from typing import Optional

from pydantic import BaseModel, Field


class AppointmentCreate(BaseModel):
    title: str = Field(min_length=1)
    description: Optional[str] = None

    appointment_date: date
    start_time: time
    end_time: time


class AppointmentUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None

    appointment_date: Optional[date] = None
    start_time: Optional[time] = None
    end_time: Optional[time] = None


class AppointmentResponse(BaseModel):
    id: int

    title: str
    description: Optional[str]

    appointment_date: date
    start_time: time
    end_time: time

    status: str

    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True